desc/objectives/utils.py

```diff
@@ -135,32 +137,61 @@ def factorize_linear_constraints(constraints, objective_args):
     A = {}
     b = {}
     Ainv = {}
+    arg_obj_lists = {}
     xp = jnp.zeros(dim_x)  # particular solution to Ax=b
     constraint_args = []  # all args used in constraints
     unfixed_args = []  # subset of constraint args for unfixed objectives
 
     # linear constraint matrices for each objective
-    for obj in constraints:
+    for obj_ind, obj in enumerate(constraints):
         if len(obj.args) > 1:
             raise ValueError("Linear constraints must have only 1 argument.")
         arg = obj.args[0]
         if arg not in objective_args:
             continue
-        constraint_args.append(arg)
+        if arg not in constraint_args:  # hope this does not mess
+            constraint_args.append(arg)
         if obj.fixed and obj.dim_f == obj.dimensions[obj.target_arg]:
             # if all coefficients are fixed the constraint matrices are not needed
```

Comment:
What would happen if the user gives 2 constraints that fix the same modes to different values here?

Reply:
Good question, the way it is coded now I think the second would overwrite the first? I only was thinking about fixing not all the modes so this conditional was not being activated, there should be some error thrown I think saying that the constraints are incompatible like I have in the case where not all modes are fixed